samples/ProjectGrandmaster.Unity/Assets/Script/Board/BoardInformation.cs

```diff
@@ -0,0 +1,633 @@
+
+            //Turn off collisions for all pieces if ghost animation is active
+            if (ghostActive)
+            {
+                Physics.IgnoreLayerCollision(14, 14);
+                Physics.IgnoreLayerCollision(15, 15);
+                Physics.IgnoreLayerCollision(14, 15);
+            }
+
+            // Disable the result display
+            EndGameResult.SetActive(false);
+            pawnPromo.SetActive(false);
+        }
+
+        public GameObject[,] GetBoard() { return board; }
+
+        public int GetTurn() { return (int)turn; }
+
+        public List<GameObject> GetPieceAvailable() { return piecesOnBoard; }
+
+        public GameObject GetBlackKing() { return blackKing; }
+
+        public GameObject GetWhiteKing() { return whiteKing; }
+
+        public LayerMask GetChessboardLayer()
+        {
+            return chessboardLayer;
+        }
+
+        /// <summary>
+        /// If Multiplayer toggled off in the menu, AI = true, and vice versa
+        /// </summary>
+        public void toggleAI()
+        {
+            if (AIEnabled) { AIEnabled = false; }
+            else { AIEnabled = true; }
+        }
+
+        /// <summary>
+        /// If ghosting enabled in the menu, Ghosting = true, and vice versa
+        /// </summary>
+        public void toggleGhosting()
+        {
+            if (ghostActive) {
+                ghostActive = false;
+
+                // Pieces ignore collisions if ghosting is on
+                Physics.IgnoreLayerCollision(14, 15, false);
+                Physics.IgnoreLayerCollision(14, 14, false);
+                Physics.IgnoreLayerCollision(15, 15, false);
+            }
+            else {
+                ghostActive = true;
+
+                // Turn collisions back on
+                Physics.IgnoreLayerCollision(14, 15, true);
+                Physics.IgnoreLayerCollision(14, 14, true);
+                Physics.IgnoreLayerCollision(15, 15, true);
+            }
+        }
+
+        /// <summary>
+        /// If White played, change to Black's turn and vice versa
+        /// </summary>
+        public void NextTurn()
+        {
+            if (turn == 0)
+            {
+                turn = Colours.Black;
+                whiteArrow.GetComponent<Renderer>().material = opponentTurn;
+                blackArrow.GetComponent<Renderer>().material = playerTurn;
+            }
+            else
+            {
+                turn = Colours.White;
+                whiteArrow.GetComponent<Renderer>().material = playerTurn;
+                blackArrow.GetComponent<Renderer>().material = opponentTurn;
+            }
+        }
+
+        public void ResetTurn() {
+            turn = Colours.Black;
+            
+            // Change arrow colours and reset turn to white
+            NextTurn();
+        }
+
+        public void PrintBoard()
+        {
+            for (int i = 7; i >= 0; i--)
+            {
+                string row = "";
```

Comment:
I presume this only don during debugging in the editor?

Also, I would still use StringBuilder for perf purposes

Reply:
Removed from script.